### PR TITLE
Fix/run vcpkg

### PR DESCRIPTION
**Pull request by jmkerloch**
- use of vcpkg as submodule instead of github action run-vcpkg
- use of github action to cache vcpkg installed package
- test of ccache use to speed up compilation on ubuntu

### CODE REVIEW

File: .github/workflows/releasesWindows.yml [hunk 1]
```diff
@@ -15,39 +15,49 @@ jobs:
...
+            
     - name: Download pre-compiled librairies
       run: |
            choco install wget
-           wget https://github.com/AntaresSimulatorTeam/antares-deps/releases/download/v1.0.1/rte-antares-deps-${{matrix.os}}-${{matrix.architecture}}-Release-solver.zip
-           Expand-Archive ./rte-antares-deps-${{matrix.os}}-${{matrix.architecture}}-Release-solver.zip -DestinationPath .
-           rm rte-antares-deps-${{matrix.os}}-${{matrix.architecture}}-Release-solver.zip
+           wget https://github.com/AntaresSimulatorTeam/antares-deps/releases/download/v1.0.1/rte-antares-deps-${{matrix.os}}-x64-Release-solver.zip
+           Expand-Archive ./rte-antares-deps-${{matrix.os}}-x64-Release-solver.zip -DestinationPath .
+           rm rte-antares-deps-${{matrix.os}}-x64-Release-solver.zip
```
Comment (flomnes) [inline .github/workflows/releasesWindows.yml]: Why default to x64 ?

Comment (jmkerloch) [reply .github/workflows/releasesWindows.yml]: There is no x86 version available of antares-deps

File: .github/workflows/releasesWindows.yml [hunk 2]
```diff
@@ -15,52 +15,73 @@ jobs:
...
+          
+    - name : Install deps with VCPKG
+      run: |
+          git submodule update --init vcpkg
+          cd vcpkg
+          ./bootstrap-vcpkg.sh
+          vcpkg install ${{matrix.vcpkgPackages}} --triplet ${{matrix.triplet}}
+          rm -rf buildtrees
+          rm -rf packages
+          rm -rf downloads
```
Comment (flomnes) [inline .github/workflows/releasesWindows.yml]: `rm -rf buildtrees packages downloads`

File: .github/workflows/windows-vcpkg.yml [hunk 3]
```diff
@@ -21,45 +21,66 @@ jobs:
...
+        path: |
+          ${{ env.VCPKG_ROOT }}
+          !${{ env.VCPKG_ROOT }}/buildtrees
+          !${{ env.VCPKG_ROOT }}/packages
+          !${{ env.VCPKG_ROOT }}/downloads
+        # The key is composed in a way that it gets properly invalidated: this must happen whenever vcpkg's Git commit id changes, or the list of packages changes. In this case a cache miss must happen and a new entry with a new key with be pushed to GitHub the cache service.
+        # The key includes: hash of the vcpkg.json file, the hash of the vcpkg Git commit id, and the used vcpkg's triplet. The vcpkg's commit id would suffice, but computing an hash out it does not harm.
+        # Note: given a key, the cache content is immutable. If a cache entry has been created improperly, in order the recreate the right content the key must be changed as well, and it must be brand new (i.e. not existing already).
+        key: |
+          ${{ hashFiles( 'vcpkg_manifest/vcpkg.json' ) }}-${{ hashFiles( '.git/modules/vcpkg/HEAD' )}}-${{ matrix.triplet }}-invalidate
```
Comment (flomnes) [inline .github/workflows/windows-vcpkg.yml:52]: **.git/modules/vcpkg/HEAD** is already a hash, maybe there is no need to apply the `hashFiles` function once more ?

File: .github/workflows/windows-vcpkg.yml [hunk 4]
```diff
@@ -21,45 +21,66 @@ jobs:
...
+          
+    - name : Install deps with VCPKG
+      run: |
+          git submodule update --init vcpkg
+          cd vcpkg
+          ./bootstrap-vcpkg.sh
+          vcpkg install ${{matrix.vcpkgPackages}} --triplet ${{matrix.triplet}}
+          rm -rf buildtrees
+          rm -rf packages
+          rm -rf downloads
```
Comment (flomnes) [inline .github/workflows/windows-vcpkg.yml]: Same as above

**Review by flomnes — COMMENTED** [review]
minor remarks

**Review by flomnes — APPROVED** [review]

**Pull request merged**